### PR TITLE
ROX-13378: Group new resources with deprecated in UI

### PR DESCRIPTION
**Pull request by ivan-degtiarenko**
## Description

1. Creates banner on `Access Control` page notifying of resource changes
2. Creates log entry similar to banner on Central startup
3. All replacing resources are marked in the UI
4. All deprecated/to-be-deprecated/to-be-removed resources are marked and show the version of deprecation
5. Added ClusterCVE resource description as it was missing
6. Guard “Integrations” by `Integration` permission

## Checklist
- [ ] Investigated and inspected CI test results
- [ ] Unit test and regression tests added
- [ ] Evaluated and added CHANGELOG entry if required
- [ ] Determined and documented upgrade steps
- [ ] Documented user facing changes (create PR based on [openshift/openshift-docs](https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))

If any of these don't apply, please comment below.

## Testing Performed

https://user-images.githubusercontent.com/78353299/199859029-9d10b9c6-0c86-46e0-975a-83cf05715370.mov

Screenshot 2022-11-04 at 00 42 24

### CODE REVIEW

File: central/main.go
Comment (rukletsov) [inline central/main.go:249]:
```suggestion
	log.Warn("The following permission resources have been replaced:\n" +
```
Also please the same in the UI banner.

File: central/main.go
Comment (rukletsov) [inline central/main.go:254]:
```suggestion
		"The following permission resources will be replaced in the upcoming versions:\n" +
```

File: ui/apps/platform/src/Containers/AccessControl/AccessControl.tsx
Comment (rukletsov) [inline ui/apps/platform/src/Containers/AccessControl/AccessControl.tsx:33]: Maybe a list for readability?

File: central/main.go
Comment (rukletsov) [inline central/main.go:256]: ClusterCVE is missing, also in the UI banner.

File: ui/apps/platform/src/constants/accessControl.ts
Comment (rukletsov) [inline ui/apps/platform/src/constants/accessControl.ts:131]: Any reason why ClusterCVE is not in this list?

**Review by rukletsov — COMMENTED** [review]
Thanks, Ivan, this looks good. We can ship once we get approval from someone's on the UI team and Yann merges his PR chain.

Could you please add a note to https://issues.redhat.com/browse/ROX-12750 that most of the the code added here should also be removed?

File: ui/apps/platform/src/Containers/AccessControl/AccessControl.tsx
Comment (ivan-degtiarenko) [reply ui/apps/platform/src/Containers/AccessControl/AccessControl.tsx:33]: When I try to use a list within this `Alert` component, it does not show up as a list. Wanted to ask @pedrottimark or @vjwilson if they know what might be the reason for this behaviour.

File: ui/apps/platform/src/constants/accessControl.ts
Comment (ivan-degtiarenko) [reply ui/apps/platform/src/constants/accessControl.ts:131]: My understanding was that `ClusterCVE` will be removed and not replaced by anything. @rhybrillou is that the case?

File: central/main.go
Comment (ivan-degtiarenko) [reply central/main.go:256]: See comment below - will `ClusterCVE` be replaced by other resource or just removed?

File: ui/apps/platform/src/constants/accessControl.ts
Comment (vjwilson) [reply ui/apps/platform/src/constants/accessControl.ts:131]: @md2119 @theencee 

How is `ClusterCVE` related to this change?

File: central/activecomponent/updater/updater_impl.go
Comment (md2119) [inline central/activecomponent/updater/updater_impl.go:26]: Probably not for now, but IMO all inits of scope contexts to fulfill internal workflows should be changed to `sac.WithGlobalAccess(context.Background())`. We know the purpose of each init `ctx` such as, building index at startup, or removing dependent resources, etc., and that these are used internally only. There is a risk, which historically has realized more than expected, that some feature begins depending on additional sac resources and insufficient context cause is not the obvious.

File: central/deployment/datastore/datastore_impl.go
Comment (md2119) [inline central/deployment/datastore/datastore_impl.go:270]: same comment as above

File: central/main.go
Comment (md2119) [inline central/main.go:250]: nit: `, and` here and below

File: central/main.go
Comment (md2119) [reply central/main.go:256]: `ClusterCVE` -> `Cluster`

File: ui/apps/platform/src/constants/accessControl.ts
Comment (md2119) [reply ui/apps/platform/src/constants/accessControl.ts:131]: `ClusterCVE` -> `Cluster`

File: ui/apps/platform/src/Containers/AccessControl/AccessControl.tsx
Comment (vjwilson) [reply ui/apps/platform/src/Containers/AccessControl/AccessControl.tsx:33]: > When I try to use a list within this `Alert` component, it does not show up as a list. Wanted to ask @pedrottimark or @vjwilson if they know what might be the reason for this behaviour.

There were were pretty strict style resets in pre-Red Hat style sheet, and we can't remove them until all the old style pages have been updated. We'd probably need to add a specific override in this file until then, to make a list show in this component.

https://github.com/stackrox/stackrox/blob/master/ui/apps/platform/src/css/trumps.css

**Review by janisz — COMMENTED** [review]
Could you split this PR?

**Comment by ivan-degtiarenko** [comment]
@janisz PR is 10 files only with pretty minor changes in them. It only appears as 144 files when there are conflicts with the base branch - AFAIU this is just how GitHub works

File: ui/apps/platform/src/Containers/AccessControl/AccessControl.tsx
Comment (rhybrillou) [inline ui/apps/platform/src/Containers/AccessControl/AccessControl.tsx:62]: In the Changelog entry from PR 3036, `ScannerBundle` is also in the list of permissions replaced by `Administration`.

**Review by rhybrillou — COMMENTED** [review]
Good job, I really like the additional information this work brings.
One minor comment regarding one missed item (probably when I gave you the list initially) in the list of resources to be replaced by Administration.

File: ui/apps/platform/src/types/roleResources.ts
Comment (pedrottimark) [inline ui/apps/platform/src/types/roleResources.ts:48]: Ivan, although not a blocker to merge, it sometimes reduces merge conflicts to add in alphabetical order instead of at the end of a list (in this case in the sub-list of resources under To-be-deprecated comment. On the other side of the coin, the goal is to add fewer resources in the future.

As a follow up, I will add a comment in this file similar to https://github.com/stackrox/stackrox/blob/master/central/role/resources/list.go#L26
```text
// KEEP THE FOLLOWING LIST SORTED IN LEXICOGRAPHIC ORDER (case-sensitive).
```

File: ui/apps/platform/src/hooks/usePermissions.ts
Comment (pedrottimark) [inline ui/apps/platform/src/hooks/usePermissions.ts:7]: Do you need to make the corresponding change in this file:
https://github.com/stackrox/stackrox/blob/master/ui/apps/platform/src/reducers/roles.js#L107-L119

File: ui/apps/platform/src/constants/accessControl.ts
Comment (pedrottimark) [inline ui/apps/platform/src/constants/accessControl.ts:135]: We will follow up with a PatternFly variable if explicit color does not work for dark mode

File: ui/apps/platform/src/Containers/Integrations/IntegrationsPage.tsx
Comment (pedrottimark) [inline ui/apps/platform/src/Containers/Integrations/IntegrationsPage.tsx:36]: What’s done is done. Was this a broken page which requires this change just before release?

Comment (ivan-degtiarenko) [reply ui/apps/platform/src/Containers/Integrations/IntegrationsPage.tsx:36]: This is related to this discussion:
https://srox.slack.com/archives/C02L9F6KR25/p1667488912824189

File: central/main.go
Comment (md2119) [inline central/main.go:257]: minor nit: `Cluster will cover ClusterCVE.`

**Review by md2119 — APPROVED** [review]
Nice work, @ivan-degtiarenko!

File: ui/apps/platform/src/Containers/AccessControl/PermissionSets/ResourceDescription.tsx
Comment (dhaus67) [inline ui/apps/platform/src/Containers/AccessControl/PermissionSets/ResourceDescription.tsx:16]: Do we have to add it within the UI as a new resource? I know we have other resources that are being leaked to the UI that are for internal use only, but why add another one?

**Review by dhaus67 — APPROVED** [review]
LGTM modulo one comment regarding the introduction of the `ClusterCVE` resource within the UI list + you have failing UI E2E tests which you should probably fix before merging.

File: ui/apps/platform/src/Containers/AccessControl/PermissionSets/ResourceDescription.tsx
Comment (ivan-degtiarenko) [reply ui/apps/platform/src/Containers/AccessControl/PermissionSets/ResourceDescription.tsx:16]: `ClusterCVE` is returned as a resource from the backend, this change just makes sure it's description is not empty

Comment (dhaus67) [reply ui/apps/platform/src/Containers/AccessControl/PermissionSets/ResourceDescription.tsx:16]: I see. It's unfortunate that we return it then in the first place, I suppose such is life then for the internal resources.

**Review by rhybrillou — APPROVED** [review]

**Review by pedrottimark — APPROVED** [review]

**Comment by ivan-degtiarenko** [comment]
As discussed with @pedrottimark, errors within `ci/prow/gke-ui-e2e-tests ` are flakes and are being investigated.

**Comment by pedrottimark** [comment]
@ivan-degtiarenko You can rebase to get #3748 which fixes gke-ui-e2e-tests failure.

**Pull request merged**
